biapol_utilities/label/_match_labels.py

```diff
@@ -42,47 +45,34 @@ def match_labels(label_image_x, label_image_y, method=intersection_over_union_ma
         Image that should serve as a reference for label-matching
     label_image_y : nd-array
         Image the labels of which should be paired with labels from imageA
-    method : str, optional
-        Pairing method to be used. The default is 'iou' (intersection over union).
-        
-        *iou (intersection over union)*: The intersection over union is used to 
-            measure the overlap between labels in both images. The label with
-            highest overlap is chosen as counterpart label
-            
-            iou_streshold: float, optional
-                Threshold value above which two labels are accepted as overlapping.
-                The default value is 0.25
+    method : callable, optional
+        Method to be used for matching the labels. This function is supposed to return a binary matrix with
+        `shape=(n+1, m+1)` corresponding to `match=1`, `no_match=0` of the `n` labels in `label_image_x` and `m` labels in
+        `label_image_y`. The default is thresholded_intersection_over_union_matrix.
+
 
     Returns
     -------
     nd-array
-        Processed version of imageB with labels corresponding to imageA.
+        Processed version of label_image_y with labels corresponding to label_image_x.
     """
     
-    if method == intersection_over_union_matrix:
-        threshold = kwargs.get('iou_threshold', 0.25)
+    method_metric = kwargs.get('metric', intersection_over_union_matrix)
+    method_filter = kwargs.get('filter', suppressed_similarity)
+    method_matching = kwargs.get('matching', max_similarity)
+    
     
     # relabel label_image_y to keep overlap matrix small
     label_image_y, _, _ = relabel_sequential(label_image_y)
     
+    # Calculate image similarity metric
+    similarity_matrix = method_metric(label_image_y.flatten(),
+                                      label_image_x.flatten())[1:,1:]
+    
+    # Filter similarity metric matrix
+    similarity_matrix = method_filter(similarity_matrix)
     
-    # Calculate image similarity matrix img_sim based on chosen method
-    img_sim = method(label_image_y, label_image_x)[1:,1:]
-    mmax = label_image_x.max()
+    # Apply matching technique
+    output = method_matching(label_image_x, label_image_y, similarity_matrix)
```

Comment:
I like this new mechanism with these three functions. But there is no documentation about it in the docstring